app/controllers/api/policies_controller.rb

```diff
@@ -3,5 +3,57 @@ class PoliciesController < BaseController
     include Subcollections::Conditions
     include Subcollections::Events
     include Subcollections::PolicyActions
+
+    def create_resource(type, _id, data = {})
+      assert_id_not_specified(data, type)
+      create_policy(data)
+    end
+
+    def edit_resource(type, id = nil, data = {})
+      raise BadRequestError, "Must specify an id for editing a #{type} resource" unless id
+      policy = resource_search(id, type, collection_class(:policies))
+      add_associations(data, policy)
+      policy.update_attributes(data)
+      policy
+    end
+
+    def delete_resource(type, id = nil, _data = nil)
+      raise BadRequestError, "Must specify an id for deleting a #{type} resource" unless id
+      policy = resource_search(id, type, MiqPolicy)
+      api_log_info("Deleting #{policy_ident(policy)}")
+      policy.destroy
+    rescue => err
+      action_result(false, err.to_s)
+    end
+
+    private
+
+    def create_policy(data)
+      policy = MiqPolicy.create!(:name        => data.delete("name"),
+                                 :description => data.delete("description"),
+                                 :towhat      => data.delete("towhat")
+                                )
```

Comment:
any of these required ?  if so they should be checked before hand and raise an appropriate BadRequestError.